app/models/question.rb

```diff
@@ -2,10 +2,33 @@ class Question < ApplicationRecord
 
   belongs_to :user
   has_many :answers, dependent: :destroy
+  has_many :topictaggings
+  has_many :topics, through: :topictaggings, dependent: :destroy
+  has_many :lecturetaggings
+  has_many :lectures, through: :lecturetaggings, dependent: :destroy
   has_one :correct_answer, ->{ where(correct: true) }, class_name: "Answer",
           foreign_key: "question_id"
 
   accepts_nested_attributes_for :answers, :correct_answer
   validates :correct_answer, presence: true
 
+	def all_topics=(names)
```

Comment:
Why are we overriding the default accessors here? This worked on my end, and I don't see anything wrong with it; im just curious